Common/XGBoost/src/main/java/org/tribuo/common/xgboost/XGBoostFeatureImportance.java

```diff
@@ -0,0 +1,253 @@
+    }
+    /**
+     * Total Cover is similar to cover, but not locally averaged by weight, and thus not skewed in the way that
+     * weight can be skewed. Returns only top numFeatures features.
+     * @return Ordered map where the keys are feature names and the value is the total gain, sorted descending
+     */
+    public LinkedHashMap<String, Double> getTotalCover(int numFeatures) {
+        return coalesceImportanceStream(getImportanceStream(TOTAL_COVER).limit(numFeatures));
+    }
+
+    /**
+     */
+    public List<XGBoostFeatureImportanceRecord> getImportances() {
+        Map<String, LinkedHashMap<String, Double>> importanceByType = Stream.of(GAIN, COVER, WEIGHT, TOTAL_GAIN, TOTAL_COVER)
+                .map(importanceType -> new AbstractMap.SimpleEntry<>(importanceType, coalesceImportanceStream(getImportanceStream(importanceType))))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        // this is already sorted by gain
+        List<String> features = new ArrayList<>(importanceByType.get(GAIN).keySet());
+        return features.stream().map(featureName -> new XGBoostFeatureImportanceRecord(featureName,
+                importanceByType.get(GAIN).get(featureName),
+                importanceByType.get(COVER).get(featureName),
+                importanceByType.get(WEIGHT).get(featureName),
+                importanceByType.get(TOTAL_GAIN).get(featureName),
+                importanceByType.get(TOTAL_COVER).get(featureName)))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * @param numFeatures number of features to return
+     * @return records of all importance metrics for each feature, sorted by gain.
+     */
+    public List<XGBoostFeatureImportanceRecord> getImportances(int numFeatures) {
+        Map<String, LinkedHashMap<String, Double>> importanceByType = Stream.of(GAIN, COVER, WEIGHT, TOTAL_GAIN, TOTAL_COVER)
+                .map(importanceType -> new AbstractMap.SimpleEntry<>(importanceType, coalesceImportanceStream(getImportanceStream(importanceType))))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        // this is already sorted by gain
+        List<String> features = new ArrayList<>(importanceByType.get(GAIN).keySet()).subList(0, Math.min(importanceByType.get(GAIN).keySet().size(), numFeatures));
+        return features.stream().map(featureName -> new XGBoostFeatureImportanceRecord(featureName,
+                importanceByType.get(GAIN).get(featureName),
+                importanceByType.get(COVER).get(featureName),
+                importanceByType.get(WEIGHT).get(featureName),
+                importanceByType.get(TOTAL_GAIN).get(featureName),
+                importanceByType.get(TOTAL_COVER).get(featureName)))
+                .collect(Collectors.toList());
+    }
+
+    public String toString() {
+        return "XGBoostFeatureImportance(" + getImportances(5).stream()
```

Comment:
That's a toString that generates an awful lot of garbage. Do we even want a toString here given it does computation?